lib/core/src/Data/Quantity.hs

```diff
@@ -134,49 +135,65 @@ instance (KnownSymbol unit, Buildable a) => Buildable (Quantity unit a) where
 
 -- | Opaque Haskell type to represent values between 0 and 100 (incl).
 newtype Percentage = Percentage
-    { getPercentage :: Word }
+    { getPercentage :: Scientific }
     deriving stock (Generic, Show, Eq, Ord)
     deriving newtype (ToJSON)
 
+-- | Desired number of digits after the decimal point for presenting the
+-- @Percentage@ type.
+percentageNumberOfFractionalDigits :: Int
+percentageNumberOfFractionalDigits = 2
+
 instance NFData Percentage
 
 instance FromJSON Percentage where
     parseJSON x = do
-        n <- parseJSON x
-        either (fail . show) return (mkPercentage @Int n)
+        r <- parseJSON x
+        either (fail . show) return (mkPercentage r)
 
 instance Bounded Percentage where
     minBound = Percentage 0
     maxBound = Percentage 100
 
-instance Enum Percentage where
-    fromEnum (Percentage p) = fromEnum p
-    toEnum = either (error . ("toEnum: " <>) . show) id . mkPercentage
-
 instance ToText Percentage where
-    toText (Percentage p) = T.pack (show p) <> "%"
+    toText (Percentage p) = T.pack (showS p) <> "%"
+      where
+        showS =
+            formatScientific
+            Fixed
+            (Just percentageNumberOfFractionalDigits)
```

Comment:
Please indent this so that it is obvious which identifiers are arguments and which are function names! Having them all at the same level is confusing.